### DISCUSSION TITLE
Make Quick Action button act immediately if only one action configured

### DISCUSSION THREAD

**Discussion by speleoluc** [osmandapp/OsmAnd#14933] [Ideas]
When only one action is attached to the Quick Action button, it should be performed right away when tapping this button (instead of showing the configured action as well as "Add Action").

Holding down the Quick Action button should be required to add another action in this case.

**Comment by vshcherb** [comment]
Yes, we have a plan for that.

**Comment by Binnette** [comment]
Tested today on OsmAnd Android 4.8.5. This bug seems to be fixed. So this issue can be closed.

With a quick action button containing only 1 action. On main screen the quick action button have the logo on the desired action. For example, my action is "add a hairdresser", so the quick action button have the hairdresser icon. When clicking this button, the action is performed directly.

Note the long press on the button allow to move the quick actions button. To add more actions, you need to do it via the settings menu.

![image](https://github.com/user-attachments/assets/e5cfc796-7ae6-4df7-bd20-e062a8c9c0d0)

**Reply by happmaoo** [reply]
you are right,solved my problem, thanks .
